dydx_withdrawal/views.py

```diff
@@ -0,0 +1,93 @@
+from dydx3 import DydxApiError
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.viewsets import GenericViewSet
+
+from components import OrderManager
+from dydx_withdrawal import (
+    SlowWithdrawalSerializer,
+    FastWithdrawalSerializer,
+    TransferSerializer,
+)
+
+from services import DydxWithdrawal
+from utilities.enums import ErrorCodes
+
+
+class Withdrawal(GenericViewSet):
+    """function slow withdrawal is responsible for withdrawing user's asset's.
+    :returns Withdrawal information.
+    """
+
+    def slow_withdrawal(self, request):
+        result = {"message": None, "error": None}
+        self.serializer_class = SlowWithdrawalSerializer
+        serializer = self.serializer_class(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        data = serializer.data
+        try:
+            db_ref = OrderManager()
+            withdarwal_ref = DydxWithdrawal()
+            withdrawal_data = withdarwal_ref.slow_withdrawal(data)
+            result["message"] = withdrawal_data["data"]["withdrawal"]
+            result["message"]["address"] = data["user_address"]
+            db_ref.store_withdrawal(result["message"])
+            return Response(result, status.HTTP_200_OK)
+        except DydxApiError or ValueError as e:
+            e = vars(e)
+            result["error"] = e["msg"]["errors"][0]["msg"]
+            error_codes = ErrorCodes
+            if error_codes.signature_error.value == result["error"]:
+                return Response(result, status.HTTP_400_BAD_REQUEST)
+            return Response(result, status.HTTP_400_BAD_REQUEST)
```

Comment:
HTTP_400_BAD_REQUEST should only be returned when the request body has some incorrect data.

Reply:
return Internal server error

Reply:
but if signature is wrong it's mean there is some  wrong in request body

Reply:
no that doesn't constitute as 400 bad req.. 400 bad req is if you send a wrong key in the requst body.. or you miss out on sending a needed key..
wrong signature is still and internal error

Reply:
But this error will come if you pass wrong position if

Reply:
And position id is passed in req body